internal/cmd/test.go

```diff
@@ -138,3 +139,20 @@ func runCMD(mainFunc func(), args []string) (err error) {
 
 	return err
 }
+
+func RunCommonCMDTests(test *testing.T, mainFunc func(), args []string, expectedExitCode int, index int) (output string, errs error) {
+	stdout, stderr, exitCode, err := RunCMDTest(test, mainFunc, args)
+	if err != nil {
+		errs = errors.Join(errs, fmt.Errorf("Case %d: CMD run returned an error: '%v'.", index, err))
```

Comment:
I don't think this should be returning errors.
Just `test.Errorf()` directly.

We also don't want to be cascading these errors.